### PR TITLE
Link errors to OTel spans

### LINKED ISSUE #1786: Link errors to OTel spans
### Problem Statement

If you use the Otel instrumentation caught errors are not associated with performance spans coming from otel

### Solution Brainstorm

Make sure errors are associated with the current otel span. See Java implementation for inspiration.

### PR DESCRIPTION
**Pull request by antonpirker**
This links Sentry errors to performance spans originated in OTel. Fixes #1786

### CODE REVIEW

File: sentry_sdk/integrations/opentelemetry/span_processor.py [hunk 1]
```diff
@@ -45,6 +47,38 @@ def __new__(cls):
...
+            otel_span = get_current_span()
+            if not otel_span:
+                return event
+
+            ctx = otel_span.get_span_context()
+            trace_id = format_trace_id(ctx.trace_id)
+            span_id = format_span_id(ctx.span_id)
+
+            invalid_span = (
+                trace_id == "00000000000000000000000000000000"
```
Comment (adinauer) [inline sentry_sdk/integrations/opentelemetry/span_processor.py:67]: Does python OTEL not have a method or at least constant for these?

File: sentry_sdk/integrations/opentelemetry/span_processor.py [hunk 2]
```diff
@@ -45,6 +47,38 @@ def __new__(cls):
 
         return cls.instance
 
+    def __init__(self):
+        # type: () -> None
+        @add_global_event_processor
+        def link_trace_context_to_error_event(event, hint):
```
Comment (adinauer) [inline sentry_sdk/integrations/opentelemetry/span_processor.py:53]: Should there be a check whether the `instrumenter` is set to `otel`?

**Review by adinauer — APPROVED** [review]
Mostly LGTM; maybe check `instrumenter` and use OTEL method/constant

File: sentry_sdk/integrations/opentelemetry/span_processor.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/opentelemetry/span_processor.py]: done

File: sentry_sdk/integrations/opentelemetry/span_processor.py (hunk 2 above)
Comment (antonpirker) [reply sentry_sdk/integrations/opentelemetry/span_processor.py:57]: yes. done.

**Comment by adinauer** [comment]
:shipit:

**Comment by antonpirker** [comment]
Tests for the event_processor are still missing. But then I think we are good

**Pull request merged**